### DISCUSSION TITLE
paddleocr开5进程并行后速度比单个进程文字识别速度慢了两三倍

### DISCUSSION THREAD

**Discussion by liyuweihuo** [PaddlePaddle/PaddleOCR#12360] [Q&A] [answered]
- 系统环境/System Environment：linux
- 版本号/Version：Paddle： 2.5.2 PaddleOCR：2.7.0.3 问题相关组件/Related components：cpu环境
- 运行指令/Command Code：
- 完整报错/Complete Error Message：单个图片识别速度慢三四倍

**Comment by tink2123** [comment, marked as answer]
请问是上述结论是如何统计的呢，可以提供一下测速代码。需要注意：测速要去除模型初始化时间。

**Comment by liyuweihuo** [comment]
测试同样五张图，单进程cpu在100左右，开5个进程，每个进程cpu就在40左右。
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/d2dfe926-24df-4628-97c9-a1bc7f30d58c)
执行函数代码如下：
<img width="827" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/e3f4a1b9-62ab-4010-8e0d-ec6b8d5ae170">
上述代码max_workers为1时的rec时间及其cpu情况如下
<img width="1029" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/36e3e58e-d4df-43c4-b863-1811a520a01b">

<img width="409" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/da836592-6f4b-4822-aeb3-f37251290625">

上述代码max_workers为5时的rec时间及其cpu情况如下
<img width="1031" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/35f6e1b7-441e-4e5a-862a-628d5cda97e3">

<img width="368" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/6096c4c6-ceb2-43be-a8fb-d40ea5f5ca9a">

**Comment by liyuweihuo** [comment]
可不可以解答一下，谢谢

**Comment by janpoo** [comment]
> 测试同样五张图，单进程cpu在100左右，开5个进程，每个进程cpu就在40左右。
> 
> ![image](https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/d2dfe926-24df-4628-97c9-a1bc7f30d58c)
> 
> 执行函数代码如下：
> 
> <img width="827" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/e3f4a1b9-62ab-4010-8e0d-ec6b8d5ae170">
> 
> 上述代码max_workers为1时的rec时间及其cpu情况如下
> 
> <img width="1029" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/36e3e58e-d4df-43c4-b863-1811a520a01b">
> 
> 
> 
> <img width="409" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/da836592-6f4b-4822-aeb3-f37251290625">
> 
> 
> 
> 上述代码max_workers为5时的rec时间及其cpu情况如下
> 
> <img width="1031" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/35f6e1b7-441e-4e5a-862a-628d5cda97e3">
> 
> 
> 
> <img width="368" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47705776/6096c4c6-ceb2-43be-a8fb-d40ea5f5ca9a">
> 
> 

图2中的执行函数代码"ocr=PaddleOCR(..."要放到外面，你这样写每次预测都会加载模型。

**Comment by liyuweihuo** [comment]
加载模型改到外边也这样，这个rec_res的时间是识别时间，成倍增长和每次多加载一次模型没关系。我是并行，max_workers设置为5相当于5个相同的1进程在并行执行啊。
